Add trip view timeline

Graywing13 opened the pull request:
- Add stuff to Trip View, namely tea cups & details section
- Move sidebar / tripview stuff out of AppInterface

Graywing13 committed:
Add coffee cup layout

Graywing13 committed:
Implement "more details" type of view

Graywing13 committed:
Refactor code out of AppInterface

- Move sidebar stuff into SideBar.js
- Move TripViewContent stuff into that file
- Make Breakfast uppercase

Graywing13 committed:
Resolve merge conflicts

Graywing13 committed:
Merge changes to branch

NLmeng commented on client/src/components/MapElement/MapElement.js:
```diff
@@ -31,7 +31,9 @@ export function MapElement({ className }) {
   return (
-    <div className={`${bgUrl} bg-cover bg-center ${className}`}>
+    <div
+      className={`${bgUrl} cursor-move bg-cover bg-center hover:brightness-95 ${className}`}
```
why is this cursor move btw?

Graywing13 replied:
Good catch, it really doesn't have to be except just as a visual indicator of "where you can drag the map" (after Andy adds mapbox, which has its own unique cursor) vs "where dragging will have no effect. 

But with the current design you can tell even if you don't have the cursor move thing so I will remove this.

Graywing13 replied:
Done

NLmeng commented on client/src/components/Timeline/TripViewContent.js:
```diff
@@ -0,0 +1,104 @@
+    return (
+      <Button
+        className={`font-bolder pointer-events-auto h-8 w-full cursor-pointer rounded-none bg-gradient-to-r py-1 transition-all duration-300 hover:from-slate-300 hover:via-transparent hover:to-slate-300 
+          ${isContentFullscreen ? '' : 'hover:h-9'}`}
```
personally, it seems kinda weird when the "see more" button pops up, maybe add a bit more delay or just dont increase the height on hover. (imo)

Graywing13 replied:
Ah okay, thanks for the feedback. I'll remove it then.

L0Lmaker replied:
+1 on this, highlight should be enough

Graywing13 replied:
Done

NLmeng reviewed (approved):
otherwise looks good

L0Lmaker committed:
Merge branch 'main' into Noreen/tripViewTimeline

L0Lmaker commented:
What are the cups for? @Graywing13

Graywing13 commented:
Each cup represents one day (i.e. it's what previously was 1 card), to follow with the running theme of Tea in the app. 

When you press into a cup, it should open up the day view for that cup.

L0Lmaker commented on client/src/temp/tripViewData.json:
```diff
@@ -0,0 +1,152 @@
+[
+  {
+    "color": "bg-red-400",
```
Can we make the cards more modern and sleek? Pastel colors feel a bit dry

This card is from the MUI website, I like the slight gradient and box shadow
<img width="392" alt="Screenshot 2023-06-04 at 10 54 11 AM" src="https://github.com/cpsc455-bugstorm/TravelersTea/assets/36722714/7f6a697a-2f07-4e69-8c4b-6d92d2bbe663">

Graywing13 replied:
Sounds good :) I will look into it.

Graywing13 replied:
that does look nice

Graywing13 replied:
Done

L0Lmaker commented on client/src/components/Timeline/TripViewContent.js:
```diff
@@ -0,0 +1,104 @@
+    setTripData(data)
+  }, [activeTripId])
+
+  const teaCups = useMemo(() => {
+    return tripData.map((item, index) => {
+      const displayNumber = index + 1
+      return (
+        <TeaCup
+          key={`overview-pin-${index}`}
+          tailwindBgColor={item.color}
+          displayNumber={displayNumber}
+          titleText={'Day ' + displayNumber}
+          locationNames={item['stages'].map((stage) => stage['locationName'])}
+        />
+      )
+    })
+  }, [tripData])
```
What are the teacups for?

L0Lmaker reviewed (changes requested):
Left some questions and inputs

L0Lmaker commented:
> Each cup represents one day (i.e. it's what previously was 1 card), to follow with the running theme of Tea in the app.
> 
> When you press into a cup, it should open up the day view for that cup.

Is there any way we can make it more functional? It feels awkward having it without any reason. It also takes up screen real estate so perhaps there's a better way to include this.

One idea is to have the cups as stage markers in the detailed view.

Graywing13 commented on client/src/components/Timeline/TripViewContent.js:
```diff
@@ -0,0 +1,104 @@
+  const renderTeacupRow = useMemo(() => {
+    return (
+      <div
+        className={`pointer-events-auto flex w-full items-end overflow-x-scroll p-8 pt-0 mac-scrollbar`}
```
Move cups down so that they are sitting on counter

Graywing13 commented:
Discussed to make a more minimalist view in a different PR (due to time constraints & num. lines diff), with a toggle to pick between the two. This is #69

Graywing13 committed:
Fix lint, incorporate feedbacks

- Also safelist all from-xx-400, to-xx-600 colors. There aren't that many and it drastically simplifies the gradient code.

Graywing13 committed:
Some color fixes

Graywing13 committed:
Reduce padding, annotate mock section

Graywing13 committed:
opt for 70% bg opacity on coloured cards

Graywing13 committed:
Fix some details view colouring

L0Lmaker reviewed (approved):
LGTM

L0Lmaker committed:
Merge branch 'main' into Noreen/tripViewTimeline
